packages/lit-html/src/lit-html.ts

```diff
@@ -1041,6 +1041,11 @@ class ChildPart implements Disconnectable {
   }
 
   _$setValue(value: unknown, directiveParent: DirectiveParent = this): void {
+    if (DEV_MODE && this.parentNode === null) {
+      throw new Error(
+        `This \`ChildPart\` has no \`parentNode\` and therefore cannot accept the value of ${value}. This likely means the element containing the part was manipulated in an unsupported way outside of Lit's control such that the part's marker nodes were ejected from DOM. For example, setting the element's \`innerHTML\` or \`textContent\` can do this.`
```

Comment:
Might be a little clearer?
```suggestion
        `The value ${value} was rendered to a ChildPart whose marker nodes are no longer in the DOM. This likely means the element containing the part was manipulated in an unsupported way outside of Lit's control such that the part's marker nodes were ejected from DOM. For example, setting the element's \`innerHTML\` or \`textContent\` can do this.`
```

Reply:
I'm not sure about including the value... It could be a complex object like a TemplateResult. Since we can't get to the source position of the template, we could probably do best by printing out the template strings.